### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Harrisonl**
Had a crack at some refactoring - would love some feedback.

### CODE REVIEW

File: lib/audited/auditor.rb [hunk 1]
```diff
@@ -115,11 +115,9 @@ def without_auditing(&block)
       def revisions(from_version = 1)
         audits = self.audits.from_version(from_version)
         return [] if audits.empty?
-        revisions = []
-        audits.each do |audit|
+        audits.inject([]) do |revisions, audit|
```
Comment (m0n9oose) [inline lib/audited/auditor.rb]:
```
audits.inject([]) do |revisions, audit|
  revisions << audit.revision
end
```

consider the following

```
audits.map(&:revision)
```

File: lib/audited/auditor.rb [hunk 2]
```diff
@@ -141,15 +139,11 @@ def audited_attributes
...
         dup.tap do |revision|
           revision.id = id
           revision.send :instance_variable_set, '@attributes', self.attributes if rails_below?('4.2.0')
-          revision.send :instance_variable_set, '@new_record', self.destroyed?
-          revision.send :instance_variable_set, '@persisted', !self.destroyed?
-          revision.send :instance_variable_set, '@readonly', false
-          revision.send :instance_variable_set, '@destroyed', false
-          revision.send :instance_variable_set, '@_destroyed', false
-          revision.send :instance_variable_set, '@marked_for_destruction', false
+          instance_variables.each { |key, value| revision.send("#{:instance_variable_set}", "@#{key}", value) }
```
Comment (m0n9oose) [inline lib/audited/auditor.rb]:
```
instance_variables.each { |key, value| revision.send("#{:instance_variable_set}", "@#{key}", value) }
```
why `send`?

```
instance_variables.each { |key, value| revision.instance_variable_set("@#{ key }", value) }
```

File: lib/audited/auditor.rb [hunk 3]
```diff
@@ -141,15 +139,11 @@ def audited_attributes
       protected
 
       def revision_with(attributes)
+        instance_variables = { "new_record" => self.destroyed?, "persisted" => !self.destroyed?, "readonly" => false, "destroyed" => false, "_destroyed" => false, "marked_for_destruction" => false }
         dup.tap do |revision|
           revision.id = id
           revision.send :instance_variable_set, '@attributes', self.attributes if rails_below?('4.2.0')
```
Comment (m0n9oose) [inline lib/audited/auditor.rb]:
```
revision.instance_variable_set(:@attributes, attributes) if rails_below?('4.2.0')
```

File: lib/audited/audit.rb [hunk 4]
```diff
@@ -140,6 +134,18 @@ def set_audit_user
     def set_request_uuid
       self.request_uuid ||= SecureRandom.uuid
     end
+
+    # Returns a hash of the changed values. "new" for new values, "old" for old values
+    def old_or_new_attributes(type)
```
Comment (tbugai) [inline lib/audited/audit.rb]: Not a fan of this.  This "extraction" refactor adds more complexity in order to dry up just a few lines.

File: lib/audited/auditor.rb (hunk 1 above)
Comment (tbugai) [inline lib/audited/auditor.rb]: +1

File: lib/audited/audit.rb [hunk 5]
```diff
@@ -26,16 +26,16 @@ class Audit < ::ActiveRecord::Base
...
-    scope :from_version,  ->(version){where(['version >= ?', version]) }
-    scope :to_version,    ->(version){where(['version <= ?', version]) }
-    scope :auditable_finder, ->(auditable_id, auditable_type){where(auditable_id: auditable_id, auditable_type: auditable_type)}
+    scope :ascending,     -> { reorder(version: :asc) }
+    scope :descending,    -> { reorder(version: :desc) }
+    scope :creates,       -> { where(action: 'create') }
+    scope :updates,       -> { where(action: 'update') }
+    scope :destroys,      -> { where(action: 'destroy') }
+
+    scope :up_until,      ->(date_or_time) { where('created_at <= ?', date_or_time) }
```
Comment (danabrit) [inline lib/audited/audit.rb]: If you're going to make the spacing changes after the arrow on lines 29-33, you should do the same on the scopes from 35-38.

File: lib/audited/auditor.rb [hunk 6]
```diff
@@ -141,15 +139,11 @@ def audited_attributes
       protected
 
       def revision_with(attributes)
+        instance_variables = { "new_record" => self.destroyed?, "persisted" => !self.destroyed?, "readonly" => false, "destroyed" => false, "_destroyed" => false, "marked_for_destruction" => false }
```
Comment (danabrit) [inline lib/audited/auditor.rb]: Not crazy about long-line hashes. This should be wrapped after each key/value pair, or else extracted into a method of its own.

Comment (Harrisonl) [inline lib/audited/auditor.rb]: Yeah I agree, I did this nearly 12 months ago and have learnt a lot more since then. Thanks for the feedback!

Comment (danabrit) [inline lib/audited/auditor.rb]: Do you still use audited? You're certainly welcome to touch up this PR. We welcome your input. :)

**Comment by thomasfedb** [comment]
Hey @Harrisonl. If you'd like to PR this (and the suggested changes above) against my fork at https://github.com/thomasfedb/audited then I'd be inclined to merge it.

**Comment by tbrisker** [comment]
This PR has been open for several years now with no activity and the main code has diverged from this branch. @Harrisonl , do you think it's still needed and would you be interested in returning to this or should we close it?

**Comment by tbrisker** [comment]
Thank you very much for all the effort you have put into this issue, and sorry it wasn't addressed in a timely manner. 
As this PR has become stale and no reply was received in 3 weeks, I'm going to close this PR for now. 
Feel free to reopen it if you wish to resume work on it at any time in the future!

**Pull request closed without merging**